### PR TITLE
Remove support for dot seperated composite keys

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s)
Towards KOL-5014

### What change does this PR introduce and why?
Removes support for dot seperated composite keys eg. `"key1.key2": value` equating to `{key1: {key2: value}}`

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/dataset/dataset.py [hunk 1]
```diff
@@ -189,28 +149,17 @@ def _to_serialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     return result[[column]]
 
 
+# Is flattening needed now???
```
Comment (consoull) [inline kolena/dataset/dataset.py]: Reminder for me to step through this function

File: kolena/dataset/dataset.py [hunk 2]
```diff
@@ -156,20 +125,10 @@ def _infer_datatype(df: pd.DataFrame) -> Union[pd.DataFrame, str]:
...
-        return id_fields
-    elif id_fields := get_id_fields_by(_FIELD_LOCATOR):
-        return id_fields
-    elif id_fields := get_id_fields_by(_FIELD_TEXT):
-        return id_fields
+def _infer_id_field(df: pd.DataFrame) -> str:
+    for field in [_FIELD_ID, _FIELD_LOCATOR, _FIELD_TEXT]:
+        if field in df.columns:
+            return field
     raise InputValidationError("Failed to infer the id_fields, please provide id_fields explicitly")
```
Comment (yifanwu-kolena) [inline kolena/dataset/dataset.py]: what's the drawback of keeping the original func name and typing signature?
i think return list[str] will be consistent with our `id_fields` defination, and we can use this function as entrypoint for any future extending

Comment (consoull) [reply kolena/dataset/dataset.py]: `id_fields` is no longer true since the function returns at most one id_field. Since this function is marked private I opted to change the name to accurately reflect its current behaviour

File: tests/unit/dataset/test_dataset.py [hunk 3]
```diff
@@ -424,13 +269,9 @@ def test__dataframe__data_type_field_not_exist() -> None:
...
             ["text"],
         ),
-        (
-            pd.DataFrame({"a.text": ["a", "b"], "b.text": ["c", "d"]}),
-            ["a.text", "b.text"],
-        ),
     ],
 )
-def test__infer_id_fields(input_df: pd.DataFrame, expected: List[str]) -> None:
+def test__infer_id_fields(input_df: pd.DataFrame, expected: str) -> None:
```
Comment (yifanwu-kolena) [inline tests/unit/dataset/test_dataset.py]: type hint is wrong, seems like mypy doesn't check tests files...

File: tests/unit/dataset/test_dataset.py [hunk 4]
```diff
@@ -440,7 +281,7 @@ def test__infer_id_fields(input_df: pd.DataFrame, expected: List[str]) -> None:
         pd.DataFrame(dict(text1=["a", "b", "c", "d"])),
     ],
 )
-def test__infer_id_fields__error(input_df: pd.DataFrame) -> None:
+def test__infer_id_field__error(input_df: pd.DataFrame) -> None:
```
Comment (yifanwu-kolena) [inline tests/unit/dataset/test_dataset.py]: might need to revert this change as well

**Review by yifanwu-kolena — APPROVED** [review]
some small issues in the tests files, approve to unblock

**Review by ly-kolena — APPROVED** [review]
